From b4a6d762d32c99d9aa4eaca2ed7f2b473a2e9578 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 27 Mar 2025 20:03:03 +0100
Subject: [PATCH 1648/1695] drivers: stm32_exti: remove API to read the IMR
 value

Remove the function to read the IMR value of a bank because
it no more compatible with EXTI driver integration in
interrupt framek and with removal of "wakeup-parent" property.

This patch is a revert of previous patch reviewed by
https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/429401

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git febbc90558d1118dffe24430e8e6c0d2ece4e842 drivers: stm32_exti: remove API to read the IMR value]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I7049166d8f78cce819443747b47bc12aa49b267c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/447279
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_exti.c         | 7 -------
 core/include/drivers/stm32_exti.h | 2 --
 2 files changed, 9 deletions(-)

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index 51f21bc85..3eaa4ddaa 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -220,13 +220,6 @@ void stm32_exti_unmask(struct stm32_exti_pdata *exti, uint32_t exti_line)
 	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-uint32_t stm32_exti_read_imr(struct stm32_exti_pdata *exti, unsigned int bank)
-{
-	assert(bank < _EXTI_BANK_NR);
-
-	return io_read32(exti->base + _EXTI_C1IMR(bank));
-}
-
 void stm32_exti_enable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % _EXTI_LINES_PER_BANK);
diff --git a/core/include/drivers/stm32_exti.h b/core/include/drivers/stm32_exti.h
index d7e6a6c94..e47276f29 100644
--- a/core/include/drivers/stm32_exti.h
+++ b/core/include/drivers/stm32_exti.h
@@ -90,6 +90,4 @@ void stm32_exti_set_gpio_port_sel(struct stm32_exti_pdata *exti, uint8_t bank,
  */
 void stm32_exti_set_tz(struct stm32_exti_pdata *exti, uint32_t exti_line);
 
-uint32_t stm32_exti_read_imr(struct stm32_exti_pdata *exti, unsigned int bank);
-
 #endif /*__STM32_EXTI_H*/
-- 
2.47.2

